markdown/10-Classical-Planning/exercises/ex_14/index.md

```diff
@@ -5,52 +5,9 @@ permalink: /planning-exercises/ex_14/
 breadcrumb: 10-Classical-Planning
 breadcrumb2: ex_14
 breadcrumb3: 10classicalPlanning
+breadcrumb4: ch10ex14
```

Comment:
Again? what's the use of breadcrumb2/3/4

Reply:
We discussed breadcrumbs in the meet one week prior to first evaluation ending when I started work on Bookmarks but was supposed to work upon Github API fetching. 
The various breadcrumbs are as follows in the new PR:
1) `breadcrumb1`: Actual Name of the chapter with index (For ex. `5-Adversarial-Search` )
2) `breadcrumb2`: Name of the exercise (For example `ex2`)
3) `breadcrumb3`: Name of the chapter without hyphens (For ex `5AdversarialSearch`)
4) `breadcrumb4`: Specific to each page and contains info about both name and exercise (For example `ch1ex1`)
5) `breadcrumb5`: Similar to breadcrumb4 (For example `ex5.1`)

Breacrumb1 was added by you last year. Breadcrumb5 is used for identifying bookmarked exercises. For example id of a question for bookmarking it is taken as ex1.1 which is same as ex1.1. `{{page.breadcrumb5 }}` is being used in the layout of the exercises for marking all the exercises, therefore, making them ready for bookmarking.
Breadcrumb4 is being used by Staticman for linking various exercises to the settings required for each exercise.` {{ page.breadcrumb4 }}` is being used in `staticman-comments.html `file in the includes folder as a link for posting comments (`https://staticmanlove.herokuapp.com/v3/entry/github/sachin10101998bot/aima-exercises/master' }}/{{ page.breadcrumb4 }})`. I added Staticman before I removed the hardcoded part and breadcrumb5 was needed for removing the hardcoded HTML part. I can remove all the breadcrumb4 and edit the Staticman configuration file to remove breadcrumb4. Breadcrumb3 and Breadcrumb2 are being used in the layout file of exercises for fetching Github API links. 

```
var chapterName = String('{{ page.breadcrumb }}');
var exerciseName = String('{{ page.breadcrumb2 }}');
var chapterNamestripped = String('{{ page.breadcrumb3 }}');
document.getElementById("link").innerHTML = "https://api.github.com/repos/sachin10101998/aima-exercises/contents/markdown/"+chapterName+"/exercises/"+exerciseName+"/answers";
document.getElementById("link2").innerHTML = "https://api.github.com/repos/sachin10101998/aima-exercises/contents/markdown/"+chapterName+"/exercises/"+exerciseName+"/answers";   

```
Since we will be shifting from Github API for fetching files to site.data.comments for fetching the files, we might not need 
breadcrumb2 and breadcrumb3. But since the present PR and previous PR for fetching content require Github API, therefore it's 
necessary to have them. Because the present file structure has answers folder inside the markdown folder.I'll remove Breadcrumb2 
and breadcrumb3 after we are done shifting the answers folder to the data folder. And if you insist I'll change the staticman 
configuration file to remove the need od breadcrumb4.

As for the part which has already been merged, as I told you already they are required for fetching the data from Github API and
without having those breadcrumbs, we won't be able to fetch the data from the required placed for rendering the submitted
answers as well as community answer.  

I can send a new PR so as to get rid of breadcrumb4 only, but until we have shifted all the answers files to the data folder, we 
can't get rid of other breadcrumbs.